lldb/source/Core/PluginManager.cpp

```diff
@@ -689,30 +702,38 @@ PluginManager::GetObjectFileCreateMemoryCallbackForPluginName(
 }
 
 Status PluginManager::SaveCore(const lldb::ProcessSP &process_sp,
-                               const FileSpec &outfile,
-                               lldb::SaveCoreStyle &core_style,
-                               llvm::StringRef plugin_name) {
-  if (plugin_name.empty()) {
+                               const lldb_private::CoreDumpOptions &options) {
+  Status error;
+  if (!options.GetOutputFile()) {
+    error.SetErrorString("No output file specified");
+    return error;
+  }
+
```

Comment:
You added `assert(process_sp)` in the object file plugins, but didn't check the process here. Add:
```
if (!process_sp) {
  error.SetErrorString("Invalid process");
  return error;
}
```